fast_llm/data/data/gpt/data.py

```diff
@@ -58,6 +66,7 @@ def __init__(
         distributed_config: DistributedConfig,
         vocab_size: int,
         max_sequence_length: int,
+        use_document_boundaries: bool | None = None,
```

Comment:
Not convinced about this new name. How about `separate_documents`? Or `separate_sequences`, to make it non-specific to language models?